media: i2c: imx477: Reset hblank on mode switch

naushir opened the pull request:
Reset the hblank control to the minimum value on every mode switch. This is to account for userland instances that do not yet control hblank, otherwise it gets set to a non-optimal value.
Signed-off-by: Naushir Patuck <naush@raspberrypi.com>

naushir committed:
media: i2c: imx477: Reset hblank on mode switch

Reset the hblank control to the minimum value on every mode switch. This is to
account for userland instances that do not yet control hblank, otherwise it
gets set to a non-optimal value.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>

6by9 commented:
I'm removing patches like this from other drivers. 
Or at least insert a 
```if (imx477->mode != mode)```
at https://github.com/raspberrypi/linux/blob/e3c6ee4b0197deeee200e3c53c3e460c64538760/drivers/media/i2c/imx477.c#L1631 so that reselecting the same mode doesn't reset it.

naushir commented:
Yes, makes sense!

naushir committed:
media: i2c: imx477: Do not unconditionally adjust hblank and vblank limits

On a mode change, only call imx477_set_framing_limits() to adjust the hblank
and vblank limits if the new mode is different from the existing mode. This
preserves any manual control values the user might have set.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>

6by9 commented:
LGTM
